test-browser/tests/runAndDeploy.js

```diff
@@ -0,0 +1,179 @@
+    browser.waitForElementPresent('*[data-id="remixIdeSidePanel"]')
+    .clickLaunchIcon('fileExplorers')
+    .addFile('Greet.sol', sources[0]['browser/Greet.sol'])
+    .clickLaunchIcon('udapp')
+    .selectAccount('0xCA35b7d915458EF540aDe6068dFe2F44E8fa733c')
+    .waitForElementPresent('*[data-id="Deploy - transact (not payable)"]')
+    .click('*[data-id="Deploy - transact (not payable)"]')
+    .pause(5000)
+    .testFunction('0xf887e3ac3143430b0c22d055eb25d234675e7f3246cb0824efc4c1437a1405d5', {
+      status: '0x1 Transaction mined and execution succeed',
+      'transaction hash': '0xf887e3ac3143430b0c22d055eb25d234675e7f3246cb0824efc4c1437a1405d5'
+    })
+  },
+
+  'Should run low level interaction (fallback function)': function (browser) {
+    browser.waitForElementPresent('*[data-id="remixIdeSidePanel"]')
+    .waitForElementPresent('*[data-id="universalDappUiTitleExpander"]')
+    .click('*[data-id="universalDappUiTitleExpander"]')
+    .waitForElementPresent('*[data-id="pluginManagerSettingsDeployAndRunLLTxSendTransaction"]')
+    .click('*[data-id="pluginManagerSettingsDeployAndRunLLTxSendTransaction"]')
+    .pause(5000)
+    .testFunction('0xfe718871ee0b4d03cdcac0e12e5b164efaf7e23ba952c07db76e62692867019b', {
+      status: '0x1 Transaction mined and execution succeed',
+      'transaction hash': '0xfe718871ee0b4d03cdcac0e12e5b164efaf7e23ba952c07db76e62692867019b'
+    })
+  },
+
+  'Should connect to Ropsten Test Network using MetaMask': function (browser) {
+    browser.waitForElementPresent('*[data-id="remixIdeSidePanel"]')
+    .setupMetamask(passphrase, password)
+    .click('.network-indicator__down-arrow')
+    .useXpath().click("//span[text()='Ropsten Test Network']")
+    .useCss().switchBrowserTab(0)
+    .refresh()
+    .waitForElementVisible('*[data-id="remixIdeIconPanel"]', 10000)
+    .click('*[data-id="landingPageStartSolidity"]')
+    .pause(5000)
+    .clickLaunchIcon('udapp')
+    .waitForElementPresent('*[data-id="settingsSelectEnvOptions"]')
+    .click('*[data-id="settingsSelectEnvOptions"] option[id="injected-mode"]')
+    .waitForElementPresent('*[data-id="settingsNetworkEnv"]')
+    .assert.containsText('*[data-id="settingsNetworkEnv"]', 'Ropsten (3) network')
+    .switchBrowserTab(2)
+    .waitForElementPresent('.page-container__footer-button:nth-of-type(2)')
+    .click('.page-container__footer-button:nth-of-type(2)')
+    .switchBrowserTab(0)
+  },
+
+  'Should deploy contract on Ropsten Test Network using MetaMask': function (browser) {
+    browser.waitForElementPresent('*[data-id="runTabSelectAccount"] option')
+    .clickLaunchIcon('fileExplorers')
+    .switchFile('browser/Greet.sol')
+    .clickLaunchIcon('udapp')
+    .waitForElementPresent('*[data-id="Deploy - transact (not payable)"]')
+    .click('*[data-id="Deploy - transact (not payable)"]')
+    .switchBrowserTab(2)
+    .waitForElementPresent('.transaction-status--unapproved')
+    .click('.transaction-status--unapproved')
```

Comment:
could you give some context why we need to click on `.transaction-status--unapproved'` ?

Reply:
`.transaction-status--unapproved` is the class of the dom element for an unapproved transaction.
![Screenshot from 2020-04-09 03-03-35](https://user-images.githubusercontent.com/23282059/78854406-c5834000-7a10-11ea-99f7-e5e422c69874.jpg)
  It links to an approval/confirmation page in metamask.
![Screenshot from 2020-04-09 03-03-47](https://user-images.githubusercontent.com/23282059/78854428-d59b1f80-7a10-11ea-83dc-b58c2cd909dc.png)